### ISSUE TITLE
run emrun on localhost / 127.0.0.1 by default

### ISSUE THREAD

**Issue by kolAflash** [emscripten-core/emscripten#22077]
Currently `emrun filename.html` listens to `0.0.0.0`. Instead it should listen to `localhost` / `127.0.0.1` by default.
_(checked version `3.1.46`)_

For security reasons debugging and testing tools like emrun should not listen to possibly publicly reachable addresses.  
Yes, there might be a firewall protecting public addresses. But it's better to pick safe defaults.
_(Yes, even the localhost address isn't extremly safe. But that [will](https://wicg.github.io/private-network-access/) [hopefully](https://developer.chrome.com/blog/private-network-access-update?hl=en) [change](https://bugzilla.mozilla.org/show_bug.cgi?id=354493) [soon](https://bugzilla.mozilla.org/show_bug.cgi?id=1481298))_

In case it's needed `emrun` already has a `--hostname` switch where you can manually set `0.0.0.0` to listen on all addresses.

**Comment by sbc100** [comment]
Seems reasonable to me.  @juj WDYT?   @kolAflash would you like to send a PR?  (Please add a ChangeLog entry if you do)

**Comment by kolAflash** [comment]
@sbc100 Can you give me a hand finding out why this failed?  
#22645

&nbsp;

_P.S._  
_I've another pull request which failed too. I'd be thankful if you may also have a look at it._  
_#22644_

**Comment by juj** [comment]
I agree, given that we do have a --hostname=0.0.0.0 option, we would now be able to default to listening to 127.0.0.1.

I would recommend that when emrun listens to localhost/127.0.0.1, that it would print a continuation to the help message in this line https://github.com/emscripten-core/emscripten/blob/f508b432f2c1b8104ce449f4175a5217cb948b74/emrun.py#L508 , something like `, if you want this server to be accessible from another device, pass --hostname=0.0.0.0 option to listen on the network.`

Otherwise people might be surprised "why doesn't this work" kind of a way, and not know what to do to make it work.
